### PR TITLE
Compute repeating flags when compiling glyphs

### PR DESCRIPTION
**Pull request by cmyr**
I'd left this as a TODO in the first patch.

### CODE REVIEW

File: write-fonts/src/tables/glyf.rs [hunk 1]
```diff
@@ -222,6 +222,57 @@ impl FontWrite for CoordDelta {
...
+        flags: impl IntoIterator<Item = SimpleGlyphFlags>,
+    ) -> impl Iterator<Item = RepeatableFlag> {
+        let mut iter = flags.into_iter();
+        let mut prev = None;
+
+        std::iter::from_fn(move || loop {
+            match (iter.next(), prev.take()) {
+                (None, prev) => return prev,
+                (Some(flag), None) => prev = Some(RepeatableFlag { flag, repeat: 0 }),
+                (Some(flag), Some(mut last)) => {
```
Comment (dfrg) [inline write-fonts/src/tables/glyf.rs:260]: The inherent logic for this is pretty nasty but this solution captures it nicely!

**Review by dfrg — APPROVED** [review]
Lgtm!

File: write-fonts/src/tables/glyf.rs [hunk 2]
```diff
@@ -295,4 +344,81 @@ mod tests {
...
+        assert_eq!(read.end_pts_of_contours(), &[3]);
+        let points = read.points().collect::<Vec<_>>();
+        assert_eq!(points[0].x, 20);
+        assert_eq!(points[0].y, -100);
+        assert_eq!(points[1].x, 25);
+        assert_eq!(points[1].y, -90);
+        assert_eq!(points[2].x, 50);
+        assert_eq!(points[2].y, -69);
+        assert_eq!(points[3].x, 80);
+        assert_eq!(points[3].y, -20);
```
Comment (rsheeter) [inline write-fonts/src/tables/glyf.rs:380]: I'm curious why we don't assert `read.points().collect::<Vec<_>>()` equals `vec![Point(..), etc]`?

File: write-fonts/src/tables/glyf.rs (hunk 1 above)
Comment (rsheeter) [reply write-fonts/src/tables/glyf.rs:260]: Ew. Also +1.

File: write-fonts/src/tables/glyf.rs [hunk 3]
```diff
@@ -295,4 +344,81 @@ mod tests {
...
         assert!(points[4].on_curve);
     }
+
+    #[test]
+    fn compile_repeatable_flags() {
+        let mut path = BezPath::new();
+        path.move_to((20., -100.));
+        path.line_to((80., -20.));
+        path.line_to((50., -69.));
+        path.line_to((25., -90.));
```
Comment (rsheeter) [inline write-fonts/src/tables/glyf.rs:354]: Optional, BezPath::from_svg("M20,-100 L80,-20 etc") might be a more concise way to write this sort of test input?

**Pull request merged**